bugwarrior/docs/services/logseq.rst

```diff
@@ -0,0 +1,150 @@
+Logseq
+======
+
+You can import `tasks <https://docs.logseq.com/#/page/tasks>`_ from `Logseq <https://logseq.com/>`_ using the ``logseq`` service name.
+
+
+Additional Requirements
+-----------------------
+
+Logseq application must active with the HTTP API server running for bugwarrior to connect.
+
+To use bugwarrior to pull tickets from Logseq you need to enable the Logseq HTTP APIs server.
+In Logseq go to **Settings** > **Features** and toggle the **HTTP APIs server** option.
+
+Next select the **API** option in the top menu to configure authorization token, e.g
+
+.. image:: pictures/logseq_token.png
+
+
+Example Service
+---------------
+
+Here's an example of a Logseq target:
+
+.. config::
+
+    [my_logseq_graph]
+    service = logseq
+    logseq.host = localhost
+    logseq.port = 12315
+    logseq.token = mybugwarrioraccesstoken
+
+The above example is the minimum required to import issues from Logseq.
+You can also feel free to use any of the configuration options described in
+:ref:`common_configuration_options` or described in `Service Features`_ below.
+
+Service Features
+----------------
+
+Task filters
+++++++++++++
+
+You can configure the service to import tasks in different states.
+By default the service will import all tasks in an active tasks states
+
+    DOING, TODO, NOW, LATER, IN-PROGRESS, WAIT, WAITING
+
+You can override this filter by setting the ``logseq.task_state`` option to a 
+comma separated list of required task states.
+
+.. config::
+    :fragment: logseq
+
+    logseq.task_state = DOING, NOW, IN-PROGRESS
```

Comment:
Simplicity, and to avoid query issues. Arbitrary datascript queries can return result sets that do not match the expected list of tasks, plus issues like above where the `logseq.DB.datascriptQuery` function doesn't match to queries that work in the Logseq app itself.

Reply:
> Simplicity...

That is arguable. If `task_state` were the only thing one might want to configure it is certainly simpler than a `query` option, but our experience with other services is that folks are going to want to configure every aspect of a query which leads to a much more complicated outcome than a single `query` parameter.

> Arbitrary datascript queries can return result sets that do not match the expected list of tasks,

Not all of our services map to things that are strictly "tasks" or "issues". For example, the Gmail service fetches emails. It seems to me that folks are likely to want to track artifacts in Logseq that are not "Logseq tasks".

> plus issues like above where the `logseq.DB.datascriptQuery` function doesn't match to queries that work in the Logseq app itself.

Is the `logseq.DB.datascriptQuery` API documented? Seems like it would be fine to say that the `query` option takes a `datascriptQuery` and link to the docs.

*BTW, I'm not hard set against your current approach, I'm just trying to draw out your rationale.*

Reply:
The API is *not* very well documented https://plugins-doc.logseq.com/logseq/DB/datascriptQuery
I got it working through trial and error.

I'd be happy to add a full query override as an advanced option, but maybe see if anyone actually asks for it first. I would suspect the simpler `task_state` selection will address 99%+ of users needs - if anyone other than me actually uses this integration :-)

Reply:
> the simpler `task_state` selection will address 99%+ of users needs

Ok, that's basically what I was hoping to hear which leads into the next question i had: Is there anything special in Logseq about the default `task_state` values you've set or are they arbitrary strings used by convention? That is, why do you suspect users will want to change this value?

Reply:
Can you explain your use case for wanting to exclude certain "open" states?

Reply:
LogSeq has two task workflow options TODO -> DOING -> DONE and LATER -> NOW -> DONE. I actually use a mix of both approaches where I tag things "I" need to do with TODO and DOING, and things I've delegated to others with NOW/LATER.  In taskwarrior I just want the pull the things I need to work on.

Reply:
Thanks, that's the last piece of the puzzle. To summarize my understanding of this thread:

- The meaning of Logseq's states is ill-defined, so we need to provide a configuration mechanism for filtering out states that users may use for unactionable tasks.
- Mastery of the poorly documented datascriptQuery language is not necessary for all Logseq users, so merely exposing the entire query as a configuration option is not very user friendly.

Time will tell if you made the right judgment but I'm satisfied that you thought through this decision, so I'm going to mark this conversation as resolved.